src/auto-complete.component.ts

```diff
@@ -314,4 +309,17 @@ export class NguiAutoCompleteComponent implements OnInit {
     };
   })();
 
+  private handleArrowKeyEvent(isUp: boolean) {
+    let totalNumItem = this.filteredList.length;
+
+    if (!totalNumItem) {
+      return;
+    }
+
+    const itemIndex = isUp ? totalNumItem + this.itemIndex - 1 : totalNumItem + this.itemIndex + 1;
```

Comment:
And this ? 
` const itemIndex = evt.keyCode === 38 ? totalNumItem + this.itemIndex - 1 : totalNumItem + this.itemIndex + 1;`